### DISCUSSION TITLE
Fonts are not loading properly & Broken links

### DISCUSSION THREAD

**Discussion by gitgilleszeimet** [thegetty/quire#532] [Get Help] [answered]
Dear Quire Community,

After testing our project on my private website, it now moved to its final destination: https://collections.mnha.lu/open/mosaik-marmor-musen/

Unfortunately, the fonts are not loading properly: Failed to load resource (404) and links from one page of the project to another page of the project do not work.

I presume that this is linked to the BaseURL, but I cannot figure out what's wrong.

Thank you for your help,

Gilles

**Comment by geealbers** [comment, marked as answer]
Hi @gitgilleszeimet, so it looks like the font paths have an extra instance of `mosaik-marmor-musen/` in them: `http://collections.mnha.lu/open/mosaik-marmor-musen/mosaik-marmor-musen/fonts/ibmplexsanscondensed-regular.woff2`.

Can you let us know the answers to these three questions so we can assist in troubleshooting?: 

1. What is the fonts `outputPath` on line 92 of your `themes/default/webpack/webpack.config.prod.js`?
2. What is the baseURL set to in `config/site.yml`?
3. Is there a `config/_default` directory in your project?

**Reply by gitgilleszeimet** [reply]
Hi @geealbers, please find my answers below:

1) outputPath: "mosaik-marmor-musen/fonts/"

2) baseURL: "https://collections.mnha.lu/open/mosaik-marmor-musen/"
     relativeURLs: ""

3) Yes, but it is empty after you told me to delete its config to fix a previous problem.

**Reply by geealbers** [reply]
Great. I'm hoping that changing `outputPath` to just "fonts/" might fix the issue. Can you give it a try and let me know?

**Reply by gitgilleszeimet** [reply]
Changing the outputPath did not fix the problem.

**Reply by geealbers** [reply]
@gitgilleszeimet I'm unable to reproduce the issue you're getting. Could you zip up your `config/` directory, the `themes/default/webpack/` directory and your config.yml file and attach them here so I can do some further testing? Or, if your project is on GitHub and you can give me access to the repo, that would work too. Thanks

**Reply by gitgilleszeimet** [reply]
@geealbers I have invited you to the repo.

**Reply by geealbers** [reply]
@gitgilleszeimet So it looks like the original fix actually worked, you may just need to do a hard refresh in your browser to get it to reload the style sheet and the fonts. (A "hard refresh" will clear the browser cache just for the page/site you're on, in Firefox and Chrome on macOS, you can do it with Command-Shift-R.) Or you can just try another browser or do a regular refresh several times.

So you know, I figured this out by looking at the font URLs in `site/css/application.css` in your repo. They were correct, and not like what was deployed to your site when you first posted this issue. So, I went back to your site and did the refresh and the fonts loaded.

**Reply by gitgilleszeimet** [reply]
Thank you and apologies for forgetting to empty the caches.

**Reply by geealbers** [reply]
You're welcome, and no apologies necessary, I did the same thing! 
Just glad it's working now.

**Reply by gitgilleszeimet** [reply]
Hi @geealbers The fonts are now getting displayed online, but when generating the PDF I get the following error messages:

gilleszeimet@Gilless-MacBook-Pro-2 mosaik-marmor-musen % quire pdf
⠇ Building PDFBrowserslist: caniuse-lite is outdated. Please run next command `npm update`
⠸ Building PDFprince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bold.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bold.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bold.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bold.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-regular.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-regular.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-regular.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-regular.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-italic.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-italic.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-italic.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-italic.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-regular.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-regular.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-regular.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-regular.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-italic.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-italic.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-italic.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-italic.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-regular.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-regular.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-regular.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-regular.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-italic.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-italic.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-italic.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-italic.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-bold.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-bold.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-bold.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notoserif-bold.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bold.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bold.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bold.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bold.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bolditalic.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bolditalic.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bolditalic.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/ibmplexsanscondensed-bolditalic.woff: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bolditalic.woff2: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bolditalic.woff2: Could not resolve host: https
prince: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bolditalic.woff: warning: http://https//collections.mnha.lu/open/mosaik-marmor-musen/fonts/notosans-bolditalic.woff: Could not resolve host: https

Thank you in advance for your help,

Gilles

**Reply by geealbers** [reply]
hi @gitgilleszeimet, though it shouldn't be an issue, I think the `baseURL` you have in the `config.yaml` file is the problem. If you change that to `baseURL: ""` and try again, the pdf should output as expected.

**Reply by gitgilleszeimet** [reply]
Hi @geealbers ! Your solution worked. Thank you.

**Reply by gitgilleszeimet** [reply]
Hi @geealbers ! The fonts are now getting displayed (site & pdf), but I still have problems with linking to an image on another page.

In preview the link to Abb. 4 works, but not on the exported site: https://collections.mnha.lu/open/mosaik-marmor-musen/04-chap-1/chap-1-02/

Again, I am most thankful for your help,

Gilles
